src/hooks/useFileManagement.ts

```diff
@@ -1,14 +1,36 @@
-import { Dispatch, SetStateAction } from "react"
+import { Dispatch, SetStateAction, useMemo } from "react"
 import { isValidFileName } from "@/lib/utils/isValidFileName"
 import {
   CodeAndPreviewState,
   CreateFileProps,
+  DeleteFileProps,
+  PackageFile,
 } from "../components/package-port/CodeAndPreview"
+import { useGlobalStore } from "./use-global-store"
+import { useToast } from "@/components/ViewPackagePage/hooks/use-toast"
+import { Package } from "fake-snippets-api/lib/db/schema"
+import { UseMutationResult } from "react-query"
+
+export function useFileManagement({
+  setCodeAndPreviewState,
+  pkg,
+  updatePackageFilesMutation,
+  openNewPackageSaveDialog,
+  refetchPackageFiles,
+  manualEditsFileContent,
+  packageFilesWithContent,
```

Comment:
see my other comment, but this should only accept package_id. I don't see any reason to have anything else